tests/e2e/utils/vmss_utils.go

```diff
@@ -0,0 +1,146 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	azcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// FindTestVMSS returns the first VMSS in the resource group,
+// assume the VMSS is in the cluster
+func FindTestVMSS(tc *AzureTestClient, rgName string) (*azcompute.VirtualMachineScaleSet, error) {
+	Logf("FindTestVMSS: start")
+
+	vmssClient := tc.createVMSSClient()
+
+	list, err := vmssClient.List(context.Background(), rgName)
+	if err != nil {
+		return nil, err
+	}
+
+	vmssList := list.Values()
+	if len(vmssList) == 0 {
+		return nil, nil
+	}
+
+	return &vmssList[0], nil
+}
+
+// ScaleVMSS scales the given VMSS
+func ScaleVMSS(tc *AzureTestClient, vmssName, rgName string, instanceCount int64) (err error) {
+	Logf("ScaleVMSS: start")
+
+	vmssClient := tc.createVMSSClient()
+
+	vmss, err := vmssClient.Get(context.Background(), rgName, vmssName)
+	if err != nil {
+		return err
+	}
+	parameters := azcompute.VirtualMachineScaleSet{
+		Location: to.StringPtr(tc.GetLocation()),
+		Sku: &azcompute.Sku{
+			Name:     vmss.Sku.Name,
+			Capacity: to.Int64Ptr(instanceCount),
+		},
+	}
+
+	Logf("ScaleVMSS: scaling VMSS %s", vmssName)
+	_, err = vmssClient.CreateOrUpdate(context.Background(), rgName, vmssName, parameters)
+	if err != nil {
+		return err
+	}
+
+	Logf("ScaleVMSS: wait the newly scaled nodes to be exposed")
+	err = waitVMSSVMCountToEqual(int(instanceCount), vmssName)
+	return err
+}
+
+// IsNodeInVMSS defines whether the node is the instance of the VMSS
+func IsNodeInVMSS(nodeName, vmssName string) bool {
+	return strings.HasPrefix(nodeName, vmssName)
+}
+
+func waitVMSSVMCountToEqual(goal int, vmssName string) error {
+	cs, err := CreateKubeClientSet()
+	if err != nil {
+		return err
+	}
+
+	err = wait.PollImmediate(poll, vmssScaleTimeout, func() (bool, error) {
+		nodes, err := GetAgentNodes(cs)
+		if err != nil {
+			return false, err
+		}
+
+		count := 0
+		for _, node := range nodes {
+			if IsNodeInVMSS(node.Name, vmssName) {
+				count++
+			}
+		}
+
+		Logf("waitVMSSVMCountToEqual: current=%d, expected=%d", count, goal)
+		return count == goal, nil
+	})
+
+	return err
+}
+
+// PollGetVMSSVMLabel gets the label of VMs in VMSS with retry
+func PollGetVMSSVMLabel(vmss *azcompute.VirtualMachineScaleSet, key string) error {
+	cs, err := CreateKubeClientSet()
+	if err != nil {
+		return err
+	}
+
+	err = wait.PollImmediate(poll, singleCallTimeout, func() (bool, error) {
+		Logf("PollGetVMSSVMLabel: getting all nodes in VMSS")
+		nodes, err := GetAgentNodes(cs)
+		if err != nil {
+			return false, err
+		}
+		var vmssInstances []v1.Node
+		for _, node := range nodes {
+			if IsNodeInVMSS(node.Name, *vmss.Name) {
```

Comment:
a better way is compare nodeName with `vm.OsProfile.ComputerName` (need list VMSS VM).